### PR TITLE
Guard additional UI thread paths in ErrorListInfoBarController

### PR DESCRIPTION
**Pull request by duncanp-sonar**
Related to #2494

### CODE REVIEW

File: src/Integration.UnitTests/LocalServices/ErrorListInfoBarControllerTests.cs [hunk 1]
```diff
@@ -279,14 +279,8 @@ public void ErrorListInfoBarController_CurrentBackgroundProcessorCancellation()
...
             var logger = new TestLogger();
-
-            var configProvider = new Mock<IConfigurationProviderService>();
-            configProvider.Setup(x => x.GetConfiguration()).Throws(new COMException("thrown by test code"));
-
-            var host = new Mock<IHost>();
-            host.As<IServiceProvider>().Setup(x => x.GetService(typeof(IConfigurationProviderService))).Returns(configProvider.Object);
-
-            var testSubject = new ErrorListInfoBarController(host.Object, Mock.Of<IUnboundProjectFinder>(), logger);
+            var host = CreateHostWithThrowingConfigProvider(new COMException("thrown by test code"));
```
Comment (duncanp-sonar) [inline src/Integration.UnitTests/LocalServices/ErrorListInfoBarControllerTests.cs:282]: Refactored the two existing tests to use a helper method

File: src/Integration.UnitTests/LocalServices/ErrorListInfoBarControllerTests.cs [hunk 2]
```diff
@@ -796,6 +783,30 @@ public void ErrorListInfoBarController_InfoBar_ClickButton_MoreThanOnce()
             bindCommandExecuted.Should().Be(1, "Expecting the command to be executed only once");
         }
 
+        [TestMethod]
```
Comment (duncanp-sonar) [inline src/Integration.UnitTests/LocalServices/ErrorListInfoBarControllerTests.cs:786]: Added critical/non-critical exception tests for the one event handler that could be easily tested.

File: src/Integration/LocalServices/ErrorListInfoBarController.cs [hunk 3]
```diff
@@ -255,38 +256,58 @@ private void UpdateRequired(string customInfoBarMessage = null)
 
         private void CurrentErrorWindowInfoBar_Closed(object sender, EventArgs e)
         {
-            this.ClearCurrentInfoBar();
+            // Called on the UI thread -> must handle exceptions
```
Comment (duncanp-sonar) [inline src/Integration/LocalServices/ErrorListInfoBarController.cs:259]: There are two other event handlers for which uncaught exceptions (will cause VS to crash - clicking on the button in the infobar, and closing the infobar.

I haven't changed the existing code, just wrapped it in a try-catch block.

I couldn't sensibly add a test for this click handler. The handler only calls one injectable object, but that object is only created after a series of stateful transitions (i.e. checking that an update is required, then creating and wiring up an infobar).

The test would either be extremely complex, or I'd have to do quite a bit of refactoring of the product code. Writing the test didn't seem like a good use of time, and changing the product code is too risky.

I've manually tested both code paths by debugging and changing the code to throw an exception at an appropriate point.

**Comment by duncanp-sonar** [comment]
The failing check is due to insufficient coverage - seven out of ten lines are covered. The other lines are not easily testable.

**Review by rita-sonarsource — APPROVED** [review]

**Pull request merged**